delft/sequenceLabelling/preprocess.py

```diff
@@ -39,27 +43,35 @@ def __init__(self,
         self.return_casing = return_casing
         self.return_features = return_features
         self.vocab_char = None
-        self.vocab_tag  = None
+        self.vocab_tag = None
         self.vocab_case = [k for k, v in case_index.items()]
         self.max_char_length = max_char_length
 
     def fit(self, X, y):
         chars = {PAD: 0, UNK: 1}
-        tags  = {PAD: 0}
+        tags = {PAD: 0}
 
-        for w in set(itertools.chain(*X)): 
-            if not self.use_char_feature:
-                continue
-            for c in w:
-                if c not in chars:
-                    chars[c] = len(chars)
+        if self.use_char_feature:
+            temp_chars = set()
+            for w in set(itertools.chain(*X)):
+                for c in w:
+                    temp_chars.add(c)
 
+            sorted_chars = sorted(temp_chars)
+            sorted_chars = {sorted_chars[idx]: len(list(sorted_chars)[0:idx]) + 2 for idx in
+                            range(0, len(sorted_chars))}
+            chars = {**chars, **sorted_chars}
+
+        temp_tags = set()
         for t in itertools.chain(*y):
-            if t not in tags:
-                tags[t] = len(tags)
+            temp_tags.add(t)
+            sorted_tags = sorted(temp_tags)
```

Comment:
apologies for the late review. It's looking good. But this seem to be at the wrong indentation level? (or is this a tabs issue?)